gdx/src/com/badlogic/gdx/scenes/scene2d/Stage.java

```diff
@@ -194,34 +193,20 @@ public void act (float delta) {
 		// Update over actors. Done in act() because actors may change position, which can fire enter/exit without an input event.
 		for (int pointer = 0, n = pointerOverActors.length; pointer < n; pointer++) {
 			Actor overLast = pointerOverActors[pointer];
-			// Check if pointer is gone.
-			if (!pointerTouched[pointer]) {
-				if (overLast != null) {
-					pointerOverActors[pointer] = null;
-					screenToStageCoordinates(tempCoords.set(pointerScreenX[pointer], pointerScreenY[pointer]));
-					// Exit over last.
-					InputEvent event = Pools.obtain(InputEvent.class);
-					event.setType(InputEvent.Type.exit);
-					event.setStage(this);
-					event.setStageX(tempCoords.x);
-					event.setStageY(tempCoords.y);
-					event.setRelatedActor(overLast);
-					event.setPointer(pointer);
-					overLast.fire(event);
-					Pools.free(event);
-				}
-				continue;
+			if (pointerTouched[pointer]) {
+				// Update the over actor for the pointer.
+				pointerOverActors[pointer] = fireEnterAndExit(overLast, pointerScreenX[pointer], pointerScreenY[pointer], pointer);
+			} else if (overLast != null) {
+				// The pointer is gone, exit the over actor for the pointer, if any.
+				pointerOverActors[pointer] = null;
+				fireExit(overLast, pointerScreenX[pointer], pointerScreenY[pointer], pointer);
```

Comment:
This looks a lot cleaner than before. I see fireExit() was moved to its own method and is also used in actorRemoved(); could it be it called anywhere else that might be able to also use that code?

Reply:
Those are the only two places firing an `exit` event. I considered a private method to obtain and configure an event, as that is done in many places, but the code is probably a little easier to read repeating the few lines of code.